### DISCUSSION TITLE
Getting connection error while launching the elastalert pods which is basically calling /opt/elastalert/run.sh

### DISCUSSION THREAD

**Discussion by gyancoupa** [jertel/elastalert2#881] [Q&A] [answered]
I am getting the below error while launching the Kubernetes pod which is calling run.sh 

`Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/elasticsearch/connection/http_requests.py", line 161, in perform_request
    response = self.session.send(prepared_request, **send_kwargs)
  File "/usr/local/lib/python3.10/site-packages/requests/sessions.py", line 645, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/requests/adapters.py", line 501, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/elastalert-create-index", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.10/site-packages/elastalert/create_index.py", line 247, in main
    create_index_mappings(es_client=es, ea_index=index, recreate=args.recreate, old_ea_index=old_index)
  File "/usr/local/lib/python3.10/site-packages/elastalert/create_index.py", line 23, in create_index_mappings
    esinfo = es_client.info()['version']
  File "/usr/local/lib/python3.10/site-packages/elasticsearch/client/utils.py", line 152, in _wrapped
    return func(*args, params=params, headers=headers, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/elasticsearch/client/__init__.py", line 286, in info
    return self.transport.perform_request(
  File "/usr/local/lib/python3.10/site-packages/elasticsearch/transport.py", line 390, in perform_request
    raise e
  File "/usr/local/lib/python3.10/site-packages/elasticsearch/transport.py", line 358, in perform_request
    status, headers_response, data = connection.perform_request(
  File "/usr/local/lib/python3.10/site-packages/elasticsearch/connection/http_requests.py", line 177, in perform_request
    raise ConnectionError("N/A", str(e), e)
elasticsearch.exceptions.ConnectionError: ConnectionError(('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))) caused by: ConnectionError(('Connection aborted.', RemoteDisconnected('Remote end closed connection without response')))`

Env variables:-

```root@clp-dev-na10001-elastalert-756fcfb7f9-vpssd:/opt/elastalert# env | grep -i elastic | egrep 'es_user|es_pas'
es_username=elastic
root@clp-dev-na10001-elastalert-756fcfb7f9-vpssd:/opt/elastalert# env | grep -i pass
es_password=********```

Config file

```rules_folder: /opt/elastalert_rules

# How often ElastAlert will query Elasticsearch
# The unit can be anything from weeks to seconds
run_every:
  minutes: 5

# ElastAlert will buffer results from the most recent
# period of time, in case some log sources are not in real time
buffer_time:
  minutes: 15

# The Elasticsearch hostname for metadata writeback
# Note that every rule can have its own Elasticsearch host
es_host: clp-dev-na10001-elasticsearch-es-http.clp-dev-na10001.svc

# The Elasticsearch port
es_port: 9200

# Sets timeout for connecting to and reading from es_host
es_conn_timeout: 180

# The AWS region to use. Set this when using AWS-managed elasticsearch
#aws_region: us-east-1

# The AWS profile to use. Use this if you are using an aws-cli profile.
# See http://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html
# for details
#profile: test

# Optional URL prefix for Elasticsearch
#es_url_prefix: elasticsearch

# Connect with TLS to Elasticsearch
use_ssl: False

# Verify TLS certificates
verify_certs: False

# CA Certificate Path
#ca_certs: '/etc/coupa/kibana/es.ca'

# GET request with body is the default option for Elasticsearch.
# If it fails for some reason, you can pass 'GET', 'POST' or 'source'.
# See http://elasticsearch-py.readthedocs.io/en/master/connection.html?highlight=send_get_body_as#transport
# for details
#es_send_get_body_as: GET

# Option basic-auth username and password for Elasticsearch

# The index on es_host which is used for metadata storage
# This can be a unmapped index, but it is recommended that you run
# elastalert-create-index to set a mapping
writeback_index: elastalert_status

# If an alert fails for some reason, ElastAlert will retry
# sending the alert until this time period has elapsed
alert_time_limit:
  days: 2`

run.sh being run from the pod

`#!/bin/sh
set -e
elastalert-create-index --config /opt/elastalert/config.yaml
elastalert --config /opt/elastalert/config.yaml "$@"```

**Comment by jertel** [comment, marked as answer]
This message is very difficult to read due to lack of proper formatting. Please see #11 and format your message as instructed. 

As far as the connectivity issues, you will need to review the Elasticsearch logs to see if there are indicators of connectivity from the ElastAlert 2 pod. Also, from the ElastAlert 2 pod you should curl the Elasticsearch pod to make sure the hostname and port are accessible between the two pods, using the options you have set in the config yaml file.

**Comment by gyancoupa** [comment]
Hi @jertel Thanks for the reply, I am able to connect to the elastic search cluster from the pod.

`telnet clp-dev-na10001-elasticsearch-es-http 9200
Trying 172.20.57.162...
Connected to clp-dev-na10001-elasticsearch-es-http.clp-dev-na10001.svc.cluster.local.
Escape character is '^]'`

Also let me check the elasticsearch logs, I corrected the format btw.

**Reply by gyancoupa** [reply]
Hi @jertel U are right I checked on elastic search side and the below error in the log seems to be the culprit.

```{"type": "server", "timestamp": "2022-06-07T12:13:31,442Z", "level": "WARN", "component": "o.e.x.s.t.n.SecurityNetty4HttpServerTransport", "cluster.name": "clp-*-**-*", "node.name": "clp-dev-***", "message": "received plaintext http traffic on an https channel, closing connection Netty4HttpChannel{localAddress=/*****:9200, remoteAddress=/*****:57194}", "cluster.uuid": "****", "node.id": "****"  }```

**Reply by jertel** [reply]
That would do it! Looks like an easy fix.
